paw-webapp/service/src/main/java/ar/edu/itba/paw/webapp/external/TwitchClientImpl.java

```diff
@@ -0,0 +1,209 @@
+package ar.edu.itba.paw.webapp.external;
+
+import ar.edu.itba.paw.webapp.interfaces.TwitchClient;
+import ar.edu.itba.paw.webapp.model.TwitchStream;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.io.IOException;
+import java.net.URI;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Concrete implementation of the {@link TwitchClient}.
+ */
+@Component
+public class TwitchClientImpl implements TwitchClient {
+
+    /**
+     * The {@link RestTemplate} to operate as an HTTP Client.
+     */
+    private final RestTemplate restTemplate;
+
+    /**
+     * The base url for the twitch API.
+     */
+    private final String baseUrl;
+
+    /**
+     * The Twitch API streams endpoint.
+     */
+    private final String streamsEndpoint;
+
+    /**
+     * The entity to be sent to get data from the Twitch API.
+     */
+    private final HttpEntity<String> requestEntity;
+
+    @Autowired
+    public TwitchClientImpl(RestTemplate restTemplate,
+                            @Value("#{environment.getRequiredProperty(\"twitch.request.base-url\")}") String baseUrl,
+                            @Value("#{environment.getRequiredProperty(\"twitch.request.streams-endpoint\")}") String streamsEndpoint,
+                            @Value("#{environment.getRequiredProperty(\"twitch.request.content-type\")}") String contentType,
+                            @Value("#{environment.getRequiredProperty(\"twitch.clientId\")}") String clientId) {
```

Comment:
(Low priority) Hay alguna razón por la cual se usa `@Value`? En [otros lados](https://github.com/juanmbellini/PowerUp/blob/108406460266f51a387ccffdc6eaebeaa6d71652/paw-webapp/webapp/src/main/java/ar/edu/itba/paw/webapp/config/WebConfig.java#L57) yo le hago `@Autowired` un `Environment` y hago `environment.getRequiredProperty(...)`. Veo que con la annotation te ahorrás tener que autowirear el `Environment`

Reply:
Aca es basicamente lo mismo. Estoy usando Spring Expression Language (Un DSL de spring) para pedirle al bean environment (el mismo que vos inyectas en WebConfig) una required property. Es exactamente lo mismo, pero en vez de yo explicitamente pedirle las cosas, lo hace el que llama a este constructor, usando el valor que provee dicha expresión. Se entiende?

Reply:
Sí, está bien. No me gusta mucho como queda 2 niveles de strings, etc. pero if it works, it works

Reply:
Me parece bastate más limpio que estar haciéndolo a mano, pero gustos son gustos.